book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+# Integration Service
+
+
+## Overview
+
+The Integration Service is composed of controllers that facilitate automated testing of content produced by the build pipelines. It is mainly responsible for integration testing capabilities.
+
+The Integration service uses the pipeline, snapshot and environment controllers to watch for Component builds, trigger and manage the testing Tekton Pipelines and create releases based on the testing outcome(s).
+
+### Goals
+
+- We need to be able to deploy a specific set of images to an environment and record the results of integration testing.
+- Given a specific set of images, we should be able to tell if they passed application validation tests.
+- We should be able to update one component image at a time in order to support 'slow loop' application validation testing.
+- We should have a capability to automatically promote sets of passed images.
+- To be SLSA compliant, we should also be able to tell what version of the build and test pipelines were used to create each component image.
+
+### High Level Workflow
+
+- When a build pipeline completes, Integration service creates a Snapshot CR representing a new collection of components that should be tested together.
+- When Integration service sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating new “ephemeral” StoneSoup Environment CRs which trigger the GitOps Service to provision environment.
+- After the environment is ready and the application snapshot has been deployed to it, the integration service tests and validates the application according to user-provided configuration. It does this by executing Tekton Pipelineruns against the ephemeral Environment.
+- When all of the test Tekton PipelineRuns have passed, Integration service triggers a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
```

Comment:
What does "static" mean and why isn't this the development environment?

Reply:
I don't think there is a set terminology on this yet, but what this basically means is that the Integration service will deploy to all Environments that are defined in the user's workspace that are:
- Not ephemeral
- Do not have a `parentEnvironment` defined

In most cases this will apply to development or staging environments.

Reply:
```suggestion
- When all of the test Tekton PipelineRuns have passed, Integration service triggers a deployment of the application to all Environments which are not labeled as "ephemeral" and which do not have a `parentEnvironment` set. For most users, this will result in a deployment to their "development" environment (or to their "staging" environment, if they do not have a dev environment.
```

Reply:
Updated